README.md

```diff
@@ -3,21 +3,88 @@
+   See `mpc.py`, `batch_reconstruction.py`, `taskprogramrunner.py`, `reed_solomon.py`
+   Our share reconstruction implementation is aggressively batched, and uses C++ and the NTL library for performance.
+   
+The bottleneck operation in MPC is generally batch reconstruction of secret-shared values.
+Here the batch operations are implemented in C++ using the NTL library, and wrapped using Cython.
+We implement both FFT-based (quasilinear) and matrix based (superlinear). (TODO: link to benchmarks about FFT vs matrix mul)
+
+3. Offline phase (generating preprocessing ingredients).
+Since the online phase consumes preprocessed random values, we need to generate these values ahead of time. 
+For a continuously running service, we need to generate such values at the same rate we consume them.
+We use a linear overhead method `RanDouSha` [BH08]. See `offline_randousha.py`
+
+For more detail on the HoneyBadgerMPC components, see [docs/subprotocols.rst](docs/subprotocols.rst).
+
+### Comparison with other MPC implementations
+Compared to oher MPC toolkit implementations (http://www.multipartycomputation.com/mpc-software,https://github.com/rdragos/awesome-mpc#software), HoneyBadgerMPC is unique in that it focuses on robustness.
+In a network of `n` server nodes, assuming at most `t<n/3` are compromised, then HonyeBadgerMPC provides confidentiality, integrity, and availability guarantees. In MPC terminology, it is asynchronous, provides active security, has linear communication overhead, and guarantees output delivery.
+Other MPC toolkits, such as SCALE-MAMBA, Viff, EMP, SPDZ, and others, do not provide guaranteed output delivery, and so if even a single node crashes they stop providing output at all.
+
+Linear communication overhead is about scaling to large network sizes. HoneyBadgerMPC implements aggressive batching and amortization, so as more server nodes are added, the communication cost per server approaches a constant). 
+
+### Comparison with zero knowledge proofs
+As a blockchain privacy layer, MPC is seen as an alternative to commitments and zero knowledge proofs, which are already widely used in cryptocurrencies such as Zcash, Monero, Ethereum, etc.
+The main difference is that with zero knowledge proofs, typically only a single party (the prover) knows the opening of the committed values (the witness), so if this party crashes, the data is no longer available.
+In MPC, the witness is secret shared, so it is guaranteed to be available as long as `n-t` of the servers are running correctly.  the secret data is guaranteed to be available for subsequent computations, where as with zero knowledge proofs 
+blockchain privacy techniques, such as Zcash and Moner oThe main difference between MPC and  that the secret data is always available, without having any individual knowing the secret.
+, and can provide availability where zero knowledge proofs get stuck
+So far, blockchain privacy has made use of Zero Knowledge proofs.
```

Comment:
It looks like there was an edit "mixup" after the sentence

> In MPC, the witness is secret shared, so it is guaranteed to be available as long as `n-t` of the servers are running correctly.

not sure what the exact intended phrasing is ... maybe something like:

The main difference between MPC and blockchain privacy techniques, such as Zcash and Monero is that the secret data is always available, without having any individual knowing the secret, and can provide availability where zero knowledge proofs get stuck. So far, blockchain privacy has made use of Zero Knowledge proofs.